### ISSUE TITLE
[16.4 AutoRec] test loss 의 역할 

### ISSUE THREAD

**Issue by kwakjeeyoon** [kwakjeeyoon/d2l-Recsys-pytorch#3]
```
# Mask the gradient during training
if autograd.is_training():
    return pred * np.sign(input)
else:
    return pred
```
위의 코드는 train을 할 시에는 rating이 있는 부분만 prediction을 인정하고 없는 부분은 0으로 만들어 rating이 있는 데이터만 backpropagate 될 수 있도록 한 코드이다. else 부분에서는 test 를 할 시에는 모든 데이터의 loss 를 계산하게 된다. 
여기서 발생하는 의문점은 rating이 있는 데이터만 학습을 하게 되는 데 rating이 없는 부분까지 test loss에 포함하게 되면, 학습되지 않은 데이터는 당연히 loss가 줄어들지 않게 되고 결과적으로 데이터가 학습이 제대로 되고 있는지, overfitting되고 있는지 감시하는 용도로 사용되는 test loss의 필요성은 사라지게 된다. 
=> 이 의문점에 동의하는지, 동의하지 않는다면 어떤 이유로 test loss가 사용되는지 토론
